core-bundle/contao/widgets/ImageSize.php

```diff
@@ -233,11 +238,44 @@ public function generate()
 		}
 
 		return sprintf(
-			'<div id="ctrl_%s" class="tl_image_size%s">%s</div>%s',
+			'<div id="ctrl_%s" class="tl_image_size%s"%s>%s</div>%s',
 			$this->strId,
 			($this->strClass ? ' ' . $this->strClass : ''),
+			$this->getStimulusAttributes($arrValues),
 			implode(' ', $arrFields),
 			$this->wizard
 		);
 	}
+
+	private function getStimulusAttributes($arrValues)
+	{
+		if ($this->wizard)
+		{
+			return '';
+		}
+
+		$ids = array_values(array_filter($arrValues, static fn ($v) => is_numeric($v)));
+
+		if (empty($ids))
+		{
+			return '';
+		}
+
+		$security = System::getContainer()->get('security.helper');
+
+		if (!$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_MODULE, 'themes') || !$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_IMAGE_SIZES))
+		{
+			return '';
+		}
+
+		$config = array(
+			'ids' => $ids,
+			'href' => System::getContainer()->get('router')->generate('contao_backend', array('do'=>'themes', 'table'=>'tl_image_size_item', 'popup'=>'1', 'nb'=>'1')),
+			'title' => $GLOBALS['TL_LANG']['MSC']['editImageSize'],
+			'icon' => Image::getUrl('edit.svg'),
+			'iconDisabled' => Image::getUrl('edit_.svg'),
+		);
+
+		return ' data-controller="contao--image-size" data-contao--image-size-config-value="' . StringUtil::specialchars(json_encode($config)) . '"';
```

Comment:
Whats an "attribute logic"?

Reply:
Giving the widget the ability to get arbitrary data attributes sounds like a concept any widget could use and not limited to Stimulus :)

Reply:
This is a good use case for the `HtmlAttributes` class - it also already handles encoding for you. I would like to add some stimulus shortcuts to it in a later PR and then we can use it here as well (will bring code completion instead of having to memorize the rather complex naming).

```php
return (new HtmlAttributes())
  ->set('data-controller', 'contao--image-size')
  ->set('data-contao--image-size-config-value', json_encode($config))
  ->toString()
;

Reply:
@Toflar no idea what you're referring to, this is a private method and each widget is generating HTML itself so a "generic" attribute feature does not make much sense to me. Also, any attribute of the widget is already added, but these attributes are coming from the widget rendering not the widget configuration.